src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_process_net_command_json.py

```diff
@@ -830,11 +834,34 @@ def on_goto_request(self, py_db, request):
 
     def on_setdebuggerproperty_request(self, py_db, request):
         args = request.arguments.kwargs
+        if 'cmd_version' in args:
+            # Default based on server process (although ideally the IDE should provide it).
+            if IS_WINDOWS:
+                ide_os = 'WINDOWS'
+            else:
+                ide_os = 'UNIX'
+            ide_os = args['cmd_version'].get('ide_os', ide_os)
+
+            # Breakpoints can be grouped by 'LINE' or by 'ID'.
+            breakpoints_by = args['cmd_version'].get("breakpoint_group_by", 'ID')
+            self.api.set_ide_os_and_breakpoints_by(py_db, 0, ide_os, breakpoints_by)
```

Comment:
I think that we should also change `debugProtocolCustom.json` and update the protocol-related classes for the arguments -- we should be accessing by attributes in the arguments (checking for `None` if it's optional) instead of checking `kwargs` in the arguments (and change those properties to `camelCase`).

Reply:
Actually, reviewing the code (again), we no longer need to set breakpoints by `id` nor `line` anymore using the DAP (this is only needed for `cmd_set_break` and is not applicable to the DAP) -- so, we can just create `self.api.set_ide_os` and stop sending the parameter relating to `id` or `line`.

Reply:
Do we need to set `py_db._set_breakpoints_with_id = True` as a default some where? or that setting not used at all in this scenario?

Reply:
It's not used at all in this scenario.